xunit-test/TranscoderTest/TranscoderTest.cpp

```diff
@@ -0,0 +1,247 @@
+#include "TranscoderTest.h"
+
+using ::infinispan::hotrod::ConfigurationBuilder;
+using ::infinispan::hotrod::Configuration;
+
+std::unique_ptr<infinispan::hotrod::RemoteCacheManager> TranscoderTest::remoteCacheManager;
+
+TranscoderTest::TranscoderTest()
+{
+}
+
+void TranscoderTest::SetUp() {
+	if (TranscoderTest::remoteCacheManager == nullptr){
+		ConfigurationBuilder builder;
+		builder.addServer().host("127.0.0.1").port(11222);
+		builder.protocolVersion(Configuration::PROTOCOL_VERSION_28);
+		builder.balancingStrategyProducer(nullptr);
+		TranscoderTest::remoteCacheManager.reset(new RemoteCacheManager(builder.build(), false));
+	}
+};
+
+void TranscoderTest::TearDown() {
+	TranscoderTest::remoteCacheManager->stop();
+}
+
+// Check transcoding between some media types
+
+
+// Write in jboss format and read raw bytes. Cache is raw bytes
+TEST_F(TranscoderTest, DefaultCacheWriteJbossReadRawBytes) {
+    TranscoderTest::remoteCacheManager->start();
+
+    JBasicMarshaller<std::string> *km = new JBasicMarshaller<std::string>();
+    JBasicMarshaller<std::string> *vm = new JBasicMarshaller<std::string>();
+    RemoteCache<std::string, std::string>& cache = TranscoderTest::remoteCacheManager->getCache<std::string, std::string>(km,
+            &Marshaller<std::string>::destroy,
+            vm,
+            &Marshaller<std::string>::destroy,
+            "default", true);
+
+    // Define a cache with a raw media type for the value
+    // and check that it returns value with the jboss marshaller preamble
+    DataFormat<std::string, std::string> df1;
+    df1.keyMediaType = MediaType(APPLICATION_JBOSS_MARSHALLING_TYPE);
+    df1.valueMediaType = MediaType(APPLICATION_UNKNOWN_TYPE);
+    df1.valueMarshaller.reset(new BasicMarshaller<std::string>());
+    RemoteCache<std::string, std::string> rawCache = cache.withDataFormat(&df1);
+
+    std::string k1("key13");
+    std::string v1("boron");
+
+    cache.clear();
+    // Store data with jboss marshaller
+    cache.put(k1,v1);
+
+    std::shared_ptr<std::string> jv1(rawCache.get(k1));
+    // Reading raw bytes the result contains the jboss preamble
+    ASSERT_EQ(*jv1, "\003>\005boron");
+
+    //Check that the initial cache works as usual
+    std::shared_ptr<std::string> result_v1(cache.get(k1));
+    ASSERT_EQ(*result_v1, "boron");
+}
+
+// Write in jboss format and read json. Cache is raw bytes
+TEST_F(TranscoderTest, DefaultCacheWriteJbossReadJson) {
+    TranscoderTest::remoteCacheManager->start();
+
+    JBasicMarshaller<std::string> *km = new JBasicMarshaller<std::string>();
+    JBasicMarshaller<std::string> *vm = new JBasicMarshaller<std::string>();
+    RemoteCache<std::string, std::string>& cache = TranscoderTest::remoteCacheManager->getCache<std::string, std::string>(km,
+            &Marshaller<std::string>::destroy,
+            vm,
+            &Marshaller<std::string>::destroy,
+            "default", true);
+
+    // Define a cache with a raw media type for the value
```

Comment:
raw -> json?

Reply:
The comment was unclear. The "store" format of the cache is defined serverside (standalone.xml), then the clients can ask a different format for the communication. Two caches are used for this test: "default" an untyped bytearray cache (raw) and "transcodingCache" a typed cache (jboss marshalled).
Then at the protocol level json format is requested via the DataFormat.MediaType field